valintaperusteet-service/src/main/java/fi/vm/sade/service/valintaperusteet/dao/impl/HakukohdeViiteDAOImpl.java

```diff
@@ -198,4 +199,52 @@ public Optional<Valintaryhma> findValintaryhmaByHakukohdeOid(String oid) {
                 .fetchFirst());
     return Optional.ofNullable(hakukohdeViite.orElse(new HakukohdeViite()).getValintaryhma());
   }
+
+  private String lastModified(
+      String tablePrefix, LocalDateTime startDatetime, LocalDateTime endDatetime) {
+    String lastModifiedField =
+        tablePrefix.isEmpty() ? "last_modified" : String.format("%s.last_modified", tablePrefix);
+    return startDatetime != null
+        ? String.format(" where %s between :startDatetime and :endDatetime", lastModifiedField)
+        : String.format(" where %s < :endDatetime", lastModifiedField);
+  }
+
+  @Override
+  public List<String> findNewOrChangedHakukohdeOids(LocalDateTime start, LocalDateTime end) {
```

Comment:
Jäin miettimään, saisiko tän kyselyn jotenkin vielä selkeämmin luettavaksi Java >15:issa olevan text block-syntaksin avulla kun normi multiline stringit on aina vähän tuskaisia tavata, mutta toi lastModified-logiikka tekee siitä kieltämättä vähän hankalaa...